shared/message/output/CompressedImage.proto

```diff
@@ -25,27 +25,85 @@ import "google/protobuf/timestamp.proto";
 import "Matrix.proto";
 import "Vector.proto";
 
+/**
+ * Contains lens and image output information.
+ */
 message CompressedImage {
+
+    /**
+     * The lens type and details about the images it makes.
+     */
     message Lens {
+
+        /**
+         * Types of projection of our lenses.
+         */
         enum Projection {
             UNKNOWN     = 0;
             RECTILINEAR = 1;
             EQUIDISTANT = 2;
             EQUISOLID   = 3;
         }
 
-        Projection projection   = 1;
-        float      focal_length = 2;
-        float      fov          = 3;
-        fvec2      centre       = 4;
+        /**
+         * The type of projection this lens is using.
+         */
+        Projection projection = 1;
+
+        /**
+         * Normalised focal length: focal length in pixels / image width.
+         */
+        float focal_length = 2;
+
+        /**
+         * The angular diameter that the lens covers (the area that light hits on the sensor). In radians.
+         */
+        float fov = 3;
+
+        /**
+         * Normalised image centre offset: pixels from centre to optical axis / image width
+         */
+        fvec2 centre = 4;
     }
 
-    uint32                    format     = 1;
-    uvec2                     dimensions = 2;
-    bytes                     data       = 3;
-    uint32                    camera_id  = 4;
-    string                    name       = 5;
-    google.protobuf.Timestamp timestamp  = 6;
-    mat4                      Hcw        = 7;
-    Lens                      lens       = 8;
+    /**
+     * Encoding of colour channels, e.g. BGGR, PRG8, RGBA, etc.
+     * see module::output::compressor::turbojpeg::Compressor::compress for full list.
```

Comment:
This isn't the best place to find a full list. This is actually a fourcc code, so a code made up of 4 numbers. So any four characters would define "a format" (if it's valid or means anything is up to the user) but in theory, you could start adding things like `h264` etc the definition is not so limited

Reply:
Is there a best place to find a full list? Should I just refer to this list as the ones which are likely to be used?

Reply:
Maybe just say 'see module::output::compressor::turbojpeg::Compressor::compress for currently supported formats'?